# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/ia32/icc/pkg_fileutils_sysmacros.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/mknod.c.orig	Fri Jun 21 18:39:46 2002
+++ ./src/mknod.c	Fri Jun 21 18:40:16 2002
@@ -27,6 +27,14 @@
 
    David MacKenzie <djm@ai.mit.edu>  */
 
+#ifdef __INTEL_COMPILER
+#  define _SYS_SYSMACROS_H 1
+#  define major(dev) ((int)(((dev) >> 8) & 0xff))
+#  define minor(dev) ((int)((dev) & 0xff))
+#  define makedev(major, minor) ((((unsigned int) (major)) << 8) \
+                                | ((unsigned int) (minor)))
+#endif
+
 #include <config.h>
 #include <stdio.h>
 #include <getopt.h>
